CNV-44934: Document storage class migration (#86954)
openshift/openshift-docs: modules/virt-migrating-storage-class-ui.adoc

// Module included in the following assemblies:
//
// * virt/managing_vms/virtual_disks/virt-migrating-storage-class.adoc

:_mod-docs-content-type: PROCEDURE

> 

Reviewer: While we do have a CNV-native single VM migration workflow (which is documented below), we also did a lot of work in MTC to enable this.  I think it is critical that we link customers to this documentation and mention that is can support more complex scenarios (ie. migrating multiple VMs in one plan, migrating a mix of stopped and running VMs, etc).  Unfortunately there is no UI-centric documentation even though MTC does have a UI and the VM workflows are supported there.

Here is a link to the merged PR that adds the VM-specific documentation to MTC: https://github.com/migtools/mig-controller/pull/1389

Reviewer: /cc @awels

Author: Thank you, @aglitke. I can see that @anarnold97 is working on documenting this within OpenShift documentation as part of #81461 which lists this feature as a Technology Preview. Is this correct? I can think about wording alongside your suggestions and add a cross-reference to this piece of documentation when it's available, will that work for you?

Reviewer 2: @jhradilek - it was classed as a tech preview for MTC 1.8.5. However, delays to my PR mean that by MTC 1.8.6 (the next release), I believe it will no longer be a tech preview. 

Thanks

Author: Thank you, @anarnold97. I was feeling a bit uneasy about redirecting users to a Technology Preview as the preferred option over a fully supported CNV feature.

Author: Hi @aglitke, please see lines 12 through 17 for the updated reference to the MTC docs. The link currently leads to the title page of the MTC documentation but as soon as pull request #81461 gets merged and publish, I am going to replace it with a link to that specific section. Could you please review my wording and let me know if it is OK?

You can migrate one or more disks attached to a virtual machine (VM) to a different storage class by using the {product-title} web console. When performing this action on a running VM, the operation of the VM is not interrupted and the data on the migrated disks remains accessible.

[NOTE]
====
With the {VirtProductName} Operator, you can only start storage class migration for one VM at the time and the VM must be running. If you need to migrate more VMs at once or migrate a mix of running and stopped VMs, consider using the link:https://docs.redhat.com/en/documentation/migration_toolkit_for_containers/{mtc-version}/html/migration_toolkit_for_containers/index[{mtc-first}].

{mtc-full} is not part of {VirtProductName} and requires separate installation.
====

.Prerequisites

* You must have a data volume or a persistent volume claim (PVC) available for storage class migration.
* The cluster must have a node available for live migration. As part of the storage class migration, the VM is live migrated to a different node.
* The VM must be running.

.Procedure

. Navigate to *Virtualization* -> *VirtualMachines* in the web console.

> . Click the Options menu {kebab} beside the virtual machine and select *Migration* -> *Storage*. You can also access this option from the *VirtualMachine details* page by selecting *Actions* -> *Migration* -> *Storage*.

Reviewer 3: Typically we want to only state one way. If you need both in this case, might consider dropping this down to its own line or something like:
```suggestion
. Click the Options menu {kebab} beside the virtual machine and select *Migration* -> *Storage*.
+
You can also access this option from the *VirtualMachine details* page by selecting *Actions* -> *Migration* -> *Storage*.
```

> . On the *Migration details* page, choose whether to migrate the entire VM storage or selected volumes only. If you click *Selected volumes*, select all disks you intend to migrate. Click *Next* to proceed.

Reviewer 3: ```suggestion
. On the *Migration details* page, choose whether to migrate the entire VM storage or selected volumes only. If you click *Selected volumes*, select any disks that you intend to migrate. Click *Next* to proceed.
```

> . On the *Destination StorageClass* page, select the storage class to migrate to from the list of available options. Click *Next* to proceed.

Reviewer 3: I can follow this, but "to migrate to from" takes a bit of work to parse. Would this work?
```suggestion
. From the list of available options on the *Destination StorageClass* page, select the storage class to migrate to. Click *Next* to proceed.
```

> . On the *Review* page, review the list of affected disks and the target storage class and click *Migrate VirtualMachine storage* to start the migration.

Reviewer 3: ```suggestion
. On the *Review* page, review the list of affected disks and the target storage class, and then click *Migrate VirtualMachine storage* to start the migration.
```

or

```suggestion
. On the *Review* page, review the list of affected disks and the target storage class. To start the migration, click *Migrate VirtualMachine storage*.
```
this could really be two steps, but I understand it's just a review and accept from the same page type thing :)
